streamlit_app/app.py

```
@@ -0,0 +1,122 @@
+    with tabs[11]:
+        st.header("Daily Streak")
+        fig = analyzer.plot_daily_streak()
+        if fig: st.pyplot(fig)
+
+    if download_zip and output_dir:
+        if st.button("Download all PNGs as ZIP"):
+            with TemporaryDirectory() as tmpdir:
+                zip_path = os.path.join(tmpdir, 'charts.zip')
+                with ZipFile(zip_path,'w') as zfp:
+                    for fname in os.listdir(output_dir):
+                        if fname.lower().endswith('.png'): zfp.write(os.path.join(output_dir, fname), arcname=fname)
```

Ruff (0.14.7)

Multiple statements on one line (colon)

(E701)